app-editors/aee/files/aee-clang-16-build.patch

```diff
@@ -0,0 +1,40 @@
+Bug: https://bugs.gentoo.org/894372
+--- a/aee.c
++++ b/aee.c
+@@ -1543,7 +1543,7 @@ struct bufr *buf_alloc()	/* allocate space for buffers		*/
+ 	temp_buf->main_buffer = FALSE;
+ 	temp_buf->edit_buffer = FALSE;
+ 	temp_buf->dos_file = FALSE;
+-	temp_buf->journ_fd = NULL;
++	temp_buf->journ_fd = 0;
+ 	return (temp_buf);
+ }
+ 
+--- a/journal.c
++++ b/journal.c
+@@ -72,6 +72,7 @@ char *jrn_vers_str = "@(#)                           journal.c $Revision: 1.41 $
+ #include "aee.h"
+ 
+ #include <time.h>
++#include <wchar.h>
+ 
+ /*
+  |	writes the contents of the line, updates the value stored in 
+@@ -310,7 +311,7 @@ char *file_name;
+ 		 |	Unable to open journal file.
+ 		 */
+ 
+-		buffer->journ_fd = NULL;
++		buffer->journ_fd = 0;
```

Comment:
Have you checked how a `0` journ_fd is handled?

Reply:
cc @x7upLime - iirc this is one of the packages you looked at too

Reply:
I remember journ_fd is not checked that much in aee..
(int) NULL and 0 for journal_fd *should* have similar effects I guess.. since is used only in calls to read/write/lseek.

journ_fd gets set to NULL during buffer initialization, or it gets voluntarily assigned to NULL instead of -1 if a call to open() fails; recover_from_journal() in journal.c does that.. Maybe having that function called could highlight any difference.. but I haven't tried.

Reply:
https://github.com/gentoo/gentoo/pull/30728/files Other patch uses -1